File: data/scripts/fetch_dataset_kaist_cvpr15.py
#!/usr/bin/python
import os
import sys
import hashlib

wget_cmd = 'wget --content-disposition --no-check-certificate '
os.system( 'wget http://multispectral.kaist.ac.kr/pedestrian/data-kaist/md5_checksum_small_files_from_KAIST.txt')
os.system( 'mkdir ../kaist-rgbt')

md5 = {}
with open( 'md5_checksum_small_files_from_KAIST.txt', 'r') as f:
	lines = [ line.rstrip('\n').split('\t') for line in f.readlines() ]
	
	for ii, line in enumerate(lines):
		file, md5, url = line
		cmd = wget_cmd + '"%s"' % url

		print '[%d/%d] Download %s' % (ii, len(lines), file)
		if os.path.exists(file):
			print 'File %s exists. Skip.' % file
		else:
			os.system(cmd)

		dstDir = file.split('.')[0]
		if 'set' in dstDir:
			_set, _vid = dstDir.split('_')
			dstDir = os.path.join('images', _set, _vid)

		if os.path.exists( dstDir ):
			print 'Extracted folder already exists. Skip.\n\n'
			continue
		else:
			print 'Check md5 checksum for %s...' % file,
			sys.stdout.flush()

			md5_download = hashlib.md5(open(file, 'rb').read()).hexdigest()
			if md5_download != md5:
				print '[Error] m5d checksum is not correct. (%s)' % file,
				continue
			print 'done.'

			print 'Extract %s...' % file,
			sys.stdout.flush()
			os.system( 'unzip -d ../kaist-rgbt -q %s' % file )
			print 'done.\n\n'
			sys.stdout.flush()

File: data/scripts/fetch_dataset_kaist_cvpr15.m
urlRoot = 'http://multispectral.kaist.ac.kr/pedestrian/data-kaist/';

urlwrite( [urlRoot 'md5_checksum_small_files_from_KAIST.txt'], ...
 'md5_checksum_small_files_from_KAIST.txt' );

fp = fopen('md5_checksum_small_files_from_KAIST.txt');
info = textscan(fp, '%s %s %s');
fclose(fp);

mkdir( '../kaist-rgbt' );

for ii = 1:length(info{1})
    if ~exist( info{1}{ii}, 'file' )
        fprintf( 'Downloading: %s\n', info{1}{ii} );
    end
    
    if system( sprintf('curl -LO %s', info{3}{ii} ) )       
        fprintf('Use matlab func. urlwrite\n');
        urlwrite( info{3}{ii}, info{1}{ii} );
    end
    
    fprintf( 'Unzip: %s\n', info{1}{ii} );    
    
    if ~system( sprintf('unzip -d ../kaist-rgbt -q %s', info{1}{ii} ) )    
        fprintf('use matlab func. unzip\n');
        unzip( info{1}{ii}, '../kaist-rgbt' );
    end

	% Skip md5 checksum
end


